src/SnippetProvider/RaygunSnippetProvider.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace SilverStripe\TagManager\SnippetProvider;
+
+use SilverStripe\Core\Environment;
+use SilverStripe\TagManager\SnippetProvider;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms;
+use SilverStripe\Core\Manifest\ModuleResourceLoader;
+
+/**
+ * A snippet provider that lets you add Raygun
+ */
+class RaygunSnippetProvider implements SnippetProvider
+{
+    public function getTitle()
+    {
+        return "Raygun crash reporting (JavaScript front-end)";
+    }
+
+    public function getParamFields()
+    {
+        $apiKeyDefined = !empty(Environment::getEnv('SS_RAYGUN_APP_KEY'));
+
+        $imgSrc = ModuleResourceLoader::singleton()->resolveURL('sminnee/tagmanager:client/img/raygun-guide.png');
+
+        $fields = [];
+        if ($apiKeyDefined) {
+            $fields[] = (new Forms\CheckboxField('ENV_APP_KEY', 'Use "silverstripe/raygun" module'))
+                ->setDescription('Choose to reuse "silverstripe/raygun" module settings. This is the default and you want this in most cases');
+        }
+
+        if ($apiKeyDefined) {
+            $apiKeyTitle = 'API Key (only if you didn\'t select the checkbox above)';
+        } else {
+            $apiKeyTitle = 'API Key';
+        }
+
+        $fields[] = (new Forms\TextField('USER_APP_KEY', $apiKeyTitle))->setDescription('Will look like alphanumeric password');
+        $fields[] = new Forms\LiteralField('HelpImage', "<p><img src=\"$imgSrc\" style=\"width: 100%; border-radius: 30px; box-shadow: 2px 2px 20px #CCC;\"></p>");
+
+        return new FieldList(...$fields);
+    }
+
+    public function getSummary(array $params)
+    {
+        return $this->getTitle();
+    }
+
+    public function getSnippets(array $params)
+    {
+        if (isset($params['ENV_APP_KEY']) && $params['ENV_APP_KEY']) {
+            $apiKey = Environment::getEnv('SS_RAYGUN_APP_KEY');
+        } elseif (isset($params['USER_APP_KEY'])) {
+            $apiKey = trim(preg_replace('/[^A-Za-z0-9_\-]/', '', $params['USER_APP_KEY']));
```

Comment:
Why are we silently sanitising this user input? If I enter a bad app key, the integration still works, and for all I know the key is valid? I suppose that's okay, but it seems like it could be a problem if I ever took that key somewhere else and expected it to work.

Reply:
yeah, that's come over from google analytics snippet as a copy paste. I'll remove it.

Reply:
Maybe keep the `trim()` sanitisation but place it at save so that users can see the results of the sanitisation?